File: presentation/src/main/java/com/somename/producthunt/сontent/mapper/ScreeenshotUrlViewModelMapper.java
package com.somename.producthunt.сontent.mapper;

import com.somename.data.repository.datasource.mapper.Mapper;
import com.somename.data.сontent.ScreenshotUrlEntity;
import com.somename.domain.model.ScreenshotUrl;
import com.somename.producthunt.сontent.ScreenshotUrlViewModel;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class ScreeenshotUrlViewModelMapper extends Mapper<ScreenshotUrl, ScreenshotUrlViewModel> {

    @Inject
    public ScreeenshotUrlViewModelMapper() {
    }

    @Override
    public ScreenshotUrlViewModel map(ScreenshotUrl screenshotUrl) {
        ScreenshotUrlViewModel screenshotUrlViewModel = new ScreenshotUrlViewModel();
        screenshotUrlViewModel.setImage850px(screenshotUrl.getImage850px());
        return screenshotUrlViewModel;
    }

    @Override
    public ScreenshotUrl reverseMap(ScreenshotUrlViewModel screenshotUrlViewModel) {
        throw new UnsupportedOperationException();
    }
}

File: presentation/src/main/java/com/somename/producthunt/сontent/mapper/ScreeenshotUrlViewModelMapper.kt
package com.somename.producthunt.сontent.mapper

import com.somename.data.repository.datasource.mapper.Mapper
import com.somename.domain.model.ScreenshotUrl
import com.somename.producthunt.сontent.ScreenshotUrlViewModel

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ScreeenshotUrlViewModelMapper @Inject
constructor() : Mapper<ScreenshotUrl, ScreenshotUrlViewModel>() {

    override fun map(value: ScreenshotUrl): ScreenshotUrlViewModel {
        val screenshotUrlViewModel = ScreenshotUrlViewModel()
        screenshotUrlViewModel.image850px = value.image850px
        return screenshotUrlViewModel
    }

    override fun reverseMap(value: ScreenshotUrlViewModel): ScreenshotUrl {
        throw UnsupportedOperationException()
    }
}
